labs/intro-to-ada/chapters/object_oriented_programming.rst

```diff
@@ -0,0 +1,373 @@
+:orphan:
+
+Object-oriented programming
+===========================
+
+:code-config:`reset_accumulator=True;accumulate_code=False`
+
+.. include:: <isopub.txt>
+
+.. role:: ada(code)
+   :language: ada
+
+.. role:: c(code)
+   :language: c
+
+.. role:: cpp(code)
+   :language: c++
+
+Simple type extension
+---------------------
+
+In this exercise, you'll work with record types containing numeric
+components. The first step is to create a record type :ada:`T_Float` with
+a floating-point component in the :ada:`Type_Extensions` package below.
+Next, you create a record type :ada:`T_Mixed` that extends the
+:ada:`T_Float` type and adds an integer component. In case of the
+:ada:`T_Mixed` type, the goal is to have both components *synchronized*:
```

Comment:
Not sure if this is just how things are done in AdaCoreU, or if this is a problem.
The "goal" is never verified - the only actual check done by main is the class derivation. Put_line could print out empty strings and the test would pass.